src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -134,23 +138,40 @@ export const FulfillmentDetails: FC<FulfillmentDetailsFormProps> = ({
   const addressVerificationUSEnabled = !!useFeatureFlag(
     "address_verification_us"
   )
+  const addressVerificationIntlEnabled = !!useFeatureFlag(
+    "address_verification_intl"
+  )
 
   const shouldVerifyAddressOnSubmit = (values: FulfillmentValues) => {
+    const enabledForAddress =
+      (values as ShipValues).attributes.country === "US"
+        ? addressVerificationUSEnabled
+        : addressVerificationIntlEnabled
+
+    const hasSavedAddresses = !!props.me.addressConnection?.edges?.length
     return (
       values.fulfillmentType === FulfillmentType.SHIP &&
-      values.attributes.country === "US" &&
-      addressVerificationUSEnabled
+      !hasSavedAddresses &&
+      enabledForAddress &&
+      addressVerificationCount < 1
     )
     // && addressFormMode === "new_address"
   }
 
   // trigger address verification by setting this to true
   const [verifyAddressNow, setVerifyAddressNow] = useState<boolean>(false)
+  const [addressVerificationCount, setAddressVerificationCount] = useState<
+    number
+  >(0)
+
+  const handleVerificationComplete = async () => {
+    setAddressVerificationCount(addressVerificationCount + 1)
+    setVerifyAddressNow(false)
+  }
 
   const handleSubmit = (values, helpers) => {
     if (shouldVerifyAddressOnSubmit(values)) {
```

Comment:
This section is the core of the functional changes to make address verification work right. `hasSavedAddresses` is now calculated twice in this file - it might be worth adding it (or the saved addresses) to the Shipping Context (bottom of index file) but because that depends on a relay object i hesitated so far.